crates/sui-core/src/traffic_controller/nodefw_client.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021, Facebook, Inc. and its affiliates
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockAddresses {
+    pub addresses: Vec<BlockAddress>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockAddress {
+    pub source_address: String,
+    pub destination_port: u16,
+    pub ttl: u64,
+}
+
+pub struct NodeFWClient {
+    client: reqwest::Client,
+    remote_fw_url: String,
+}
+
+impl NodeFWClient {
+    pub fn new(remote_fw_url: String) -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            remote_fw_url,
+        }
+    }
+
+    pub async fn block_addresses(&self, addresses: BlockAddresses) -> Result<(), reqwest::Error> {
+        self.client
+            .post(&format!("{}/block_addresses", self.remote_fw_url))
+            .json(&addresses)
+            .send()
+            .await?;
+        Ok(())
+    }
+
+    pub async fn list_addresses(&self) -> Result<BlockAddresses, reqwest::Error> {
+        self.client
+            .get(&format!("{}/list_addresses", self.remote_fw_url))
+            .send()
+            .await?
+            .json::<BlockAddresses>()
```

Comment:
if json::<BlockAddress> cannot be deserialized correctly, does it panic here or will it be graceful and bubble out?  I think, could be wrong, that you might need to do an err check here to be safe.

Reply:
This will return an error, not panic. And I don't need to do an error check here because I'm returning the error, so delegating the check to the caller (which will have to do the check, otherwise linting will fail)